src/nvim/path.c

```diff
@@ -612,8 +612,9 @@ static size_t do_path_expand(garray_T *gap, const char_u *path,
   *s = NUL;
 
   Directory dir;
-  if (os_scandir(&dir, *buf == NUL ? "." : (char *)buf)
-      || os_isdir(*buf == NUL ? (char_u *)"." : (char_u *)buf)) {
+  if (os_file_is_readable(*buf == NUL ? "." : (char *)buf)
+      && (os_scandir(&dir, *buf == NUL ? "." : (char *)buf)
+          || os_isdir(*buf == NUL ? (char_u *)"." : buf))) {
```

Comment:
`*buf == NUL ? "." : (char *)buf` could be evaluated before the `if` to make it more readable.